### PR TITLE
cmd/operator-sdk/*: don't log usage for non-usage-related errors

### LINKED ISSUE #2674: build failure should not result in usage text being logged to console
## Bug Report

**What did you do?**

When running `operator-sdk build`, if there is a Go compilation error (for example), the `operator-sdk` console output will log the Go errors, but then also print the usage of the command.

To reproduce, make a Go source file change that yields a compilation error, such as declaring a variable but not using it.

**What did you expect to see?**

A build error should log Go compilation errors, but should not print usage information for the `operator-sdk build` command.

**What did you see instead? Under which circumstances?**

```
$ operator-sdk build <image>
...
Error: failed to build operator binary: failed to exec []string{"go", "build", "-o", "...", "-gcflags", "all=-trimpath=...", "-asmflags", "all=-trimpath=...", "..."}: exit status 2
Usage:
  operator-sdk build <image> [flags]

Flags:
      --go-build-args string      Extra Go build arguments as one string such as "-ldflags -X=main.xyz=abc"
  -h, --help                      help for build
      --image-build-args string   Extra image build arguments as one string such as "--build-arg https_proxy=$https_proxy"
      --image-builder string      Tool to build OCI images. One of: [docker, podman, buildah] (default "docker")

Global Flags:
      --verbose   Enable verbose logging
```

**Environment**
* operator-sdk version:

```
$ operator-sdk version
operator-sdk version: "v0.15.2", commit: "ffaf278993c8fcb00c6f527c9f20091eb8dd3352", go version: "go1.13.3 linux/amd64"
```

* go version:

```
$ go version
go version go1.13.8 linux/amd64
```

* Kubernetes version information:

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-18T23:30:10Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.7", GitCommit:"6c143d35bb11d74970e7bc0b6c45b6bfdffc0bd4", GitTreeState:"clean", BuildDate:"2019-12-11T12:34:17Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: 

Multi-node Kubernetes cluster built with `kubeadm`.

* Are you writing your operator in ansible, helm, or go?

Go

**Possible Solution**

If a Go compilation error is detected, avoid printing the usage info of the `operator-sdk build` command.

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by joelanford**
Closes #2674

### CODE REVIEW

**Comment by fabianvf** [comment]
Should we do this for other commands as well? Usage should probably only be printed when usage is incorrect, not when the commands fail. Doesn't need to block this merge though

/lgtm

**Comment by joelanford** [comment]
Yeah good point. We should double check the other commands. I think it's somewhat spotty at the moment.

File: cmd/operator-sdk/bundle/create.go [hunk 1]
```diff
@@ -21,10 +21,10 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/operator-framework/operator-registry/pkg/lib/bundle"
```
Comment (estroz) [inline cmd/operator-sdk/bundle/create.go]: Move this back below

File: cmd/operator-sdk/generate/k8s.go [hunk 2]
```diff
@@ -17,6 +17,8 @@ package generate
 import (
 	"fmt"
 
+	log "github.com/sirupsen/logrus"
```
Comment (estroz) [inline cmd/operator-sdk/generate/k8s.go]: Move below

File: cmd/operator-sdk/generate/openapi.go [hunk 3]
```diff
@@ -17,6 +17,8 @@ package generate
 import (
 	"fmt"
 
+	log "github.com/sirupsen/logrus"
```
Comment (estroz) [inline cmd/operator-sdk/generate/openapi.go]: Ditto

File: cmd/operator-sdk/printdeps/cmd.go [hunk 4]
```diff
@@ -17,6 +17,8 @@ package printdeps
 import (
 	"fmt"
 
+	log "github.com/sirupsen/logrus"
```
Comment (estroz) [inline cmd/operator-sdk/printdeps/cmd.go]: Ditto

**Pull request merged**